jupyterlab_git/git.py

```diff
@@ -85,19 +94,32 @@ def call_subprocess(
         output, error = process.communicate()
         return (process.returncode, output.decode("utf-8"), error.decode("utf-8"))
 
-    if username is not None and password is not None:
-        code, output, error = await call_subprocess_with_authentication(
-            cmdline,
-            username,
-            password,
-            cwd,
-            env,
-        )
-    else:
-        current_loop = tornado.ioloop.IOLoop.current()
-        code, output, error = await current_loop.run_in_executor(
-            None, call_subprocess, cmdline, cwd, env
-        )
+    await execution_lock.acquire(timeout=datetime.timedelta(seconds=MAX_WAIT_FOR_EXECUTE_S))
```

Comment:
```suggestion
    try:
        await execution_lock.acquire(timeout=datetime.timedelta(seconds=MAX_WAIT_FOR_EXECUTE_S))
    except  tornado.util.TimeoutError:
        return (1, "", "Unable to get the lock on the directory")
```

We should return a proper error message if the lock can not be acquired.